src/Page/ExpressivePage.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+namespace Zend\Navigation\Page;
+
+use Zend\Expressive\Router\Exception\RuntimeException as RouterException;
+use Zend\Expressive\Router\RouteResult;
+use Zend\Expressive\Router\RouterInterface;
+use Zend\Navigation\Exception;
+
+/**
+ * Class ExpressivePage
+ */
+class ExpressivePage extends AbstractPage
+{
+    /**
+     * Route name
+     *
+     * @var string
+     */
+    private $routeName;
+
+    /**
+     * Route parameters
+     *
+     * @var array
+     */
+    private $routeParams = [];
+
+    /**
+     * @var array
+     */
+    private $queryParams = [];
+
+    /**
+     * @var RouteResult
+     */
+    private $routeResult;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var string
+     */
+    private $hrefCache;
+
+    /**
+     * @inheritdoc
+     */
+    public function isActive($recursive = false)
+    {
+        if (! $this->active && $this->routeName !== null
+            && $this->routeResult instanceof RouteResult
+        ) {
+            $intersectionOfParams = array_intersect_assoc(
+                $this->routeResult->getMatchedParams(),
+                $this->routeParams
+            );
+
+            $matchedRouteName = $this->routeResult->getMatchedRouteName();
+
+            if ($matchedRouteName === $this->routeName
+                && count($intersectionOfParams) === count($this->routeParams)
+            ) {
+                $this->active = true;
+
+                return $this->active;
+            }
+        }
+
+        return parent::isActive($recursive);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getHref()
+    {
+        // User cache?
+        if ($this->hrefCache) {
+            return $this->hrefCache;
+        }
+
+        if ($this->routeName === null) {
+            return $this->generateUriFromResult(
+                $this->routeParams,
+                $this->routeResult
+            );
+        }
+
+        // Generate the route
+        $href = $this->router->generateUri(
+            $this->routeName,
+            $this->routeParams
+        );
+
+        // Append query parameters if there are any
+        if (count($this->queryParams) > 0) {
+            $href .= '?' . http_build_query($this->queryParams);
+        }
+
+        // Append the fragment identifier
+        if ($this->getFragment() !== null) {
+            $href .= '#' . $this->getFragment();
+        }
+
+        return $this->hrefCache = $href;
+    }
+
+    /**
+     * @param string|null $route
```

Comment:
Because in PHP7.1 we have `void !== null`
http://php.net/manual/en/migration71.new-features.php#migration71.new-features.void-functions

I wouldn't quote coding standard from ZF1 ;-)